### ISSUE TITLE
Drop ucx.pc

### ISSUE THREAD

**Issue by jengelh** [openucx/ucx#1599]
I recommend that `ucx.pc` be dropped from the sources.IMO, a program wishing to some or all of UCX can easily and _correctly_ specify all its dependencies by
```
PKG_CHECK_MODULES([ucx], [ucs uct ucp])
```
which is seriously no more typing than
```
PKG_CHECK_MODULES([ucx], [ucx])
```
At the same time, the end-user program is very clear about the exact requirements and does not rely on a variadic list (which ucx.pc is) of libraries which either contains too many libraries, or too few.

**Comment by shamisp** [comment]
I have no objection. @yosefe do u care ?

**Comment by yosefe** [comment]
fine by me

**Comment by shamisp** [comment]
Fine for me either ...

**Comment by shamisp** [comment]
@miked-mellanox Mike, do you care ?

**Comment by miked-mellanox** [comment]
But proposed macro only checks if ucx.pc file is available and pck-config can find and use it.
How dropping uxc.pc file can be repaced by check macro?

**Comment by yosefe** [comment]
looking at this again, we still want to keep ucx.pc, for example to support static link
`pkg-config --link --static ucx`

**Comment by jengelh** [comment]
But what's the point of that? ucx.pc does not distinguish between shared and static.
```
# pkg-config --link --static ucx
Unknown option --link
# pkg-config --libs --static ucx
-lucs -luct -lucp
# pkg-config --libs ucx
-lucs -luct -lucp
```
uct's dependency on e.g. ucs should be listed in `uct.pc`, not through the hammer that `ucx.pc` is, and with a `Requires`.

**Comment by yosefe** [comment]
It is true ucx.pc does not support it now, but we want to add such support, therefore we want to keep it.
Do you mean the request is not "drop ucx.pc" but "drop ucx.pc and replace it by ucs.pc, uct.pc, ucp.pc"?

**Comment by jengelh** [comment]
Yeah, my initial PKG_CHECK line in this report referenced a "ucs", so there has to be a ucs.pc at some point ;-)
